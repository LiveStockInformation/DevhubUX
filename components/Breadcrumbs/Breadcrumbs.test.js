import * as React from 'react'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'
import { Breadcrumbs } from '.'

afterEach(cleanup)

describe('Breadcrumbs component', () => {
  it('renders without crashing', () => {
    const { asFragment } = render(<Breadcrumbs items={[{ title: 'test' }]} />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('renders nothing if 0 items', () => {
    const { asFragment } = render(<Breadcrumbs />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('renders a back link if item has backlink property', () => {
    const { getByText } = render(<Breadcrumbs items={[{ title: 'test', back: true }]} />)
    const backlink = getByText('Back')
    expect(backlink).toBeInTheDocument()
  })
})
