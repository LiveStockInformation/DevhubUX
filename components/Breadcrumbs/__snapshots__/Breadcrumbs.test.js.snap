// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs component renders nothing if 0 items 1`] = `<DocumentFragment />`;

exports[`Breadcrumbs component renders without crashing 1`] = `
<DocumentFragment>
  <div
    aria-label="breadcrumb navigation"
    class="ls-breadcrumbs"
    role="navigation"
  >
    <ol
      class="ls-breadcrumbs__list"
    >
      <li
        class="ls-breadcrumbs__list-item"
      >
        <a
          class="ls-breadcrumbs__link"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        aria-current="page"
        class="ls-breadcrumbs__list-item"
      >
        test 
      </li>
    </ol>
  </div>
</DocumentFragment>
`;
